classes/module/Module.php

```diff
@@ -1269,19 +1269,11 @@ public static function getModulesOnDisk($use_config = false, $logged_on_addons =
                         $file = substr($file, 0, -2);
                     }
 
-                    // We check any parse error before including the file.
-                    // If (false) is a trick to not load the class with "eval".
-                    // This way require_once will works correctly
-                    // But namespace and use statements need to be removed
-                    $content = preg_replace('/\n[\s\t]*?use\s.*?;/', '', $file);
-                    $content = preg_replace('/\n[\s\t]*?namespace\s.*?;/', '', $content);
                     try {
-                        if (substr(`php -l $file_path`, 0, 16) == 'No syntax errors' || eval('if (false){	'.$content.' }') !== false) {
-                            require_once(_PS_MODULE_DIR_.$module.'/'.$module.'.php');
-                        } else {
-                            throw new ParseError("Parse error");
-                        }
-                    } catch (ParseError $e) {
+                        $fileContents = _PS_MODULE_DIR_.$module.'/'.$module.'.php';
+                        $parser = (new PhpParser\ParserFactory)->create(PhpParser\ParserFactory::PREFER_PHP5);
```

Comment:
This is a new feature brought to a new major version of PrestaShop. I hope people would at the same time upgrade their PHP version. 😄 
I'm pretty sure we will forget to change the constant later. Because it won't break anything, I would prefer to write directly the latest available PHP version.